Content.Server/Radiation/Systems/RadiationSystem.GridCast.cs

```diff
@@ -64,6 +68,15 @@ private void UpdateGridcast()
                     rads += ray.Rads;
             }
 
+            // Consider if the destination entity is hidden within a radiation blocking container
+            if (_container.TryFindComponentsOnEntityContainerOrParent(dest.Owner, blockers, out var comps))
+            {
+                rads -= comps.Sum(x => x.RadResistance);
+
+                if (rads <= 0)
+                    rads = 0;
+            }
```

Comment:
instead of duping this code twice, simply add a method for getting adjusted radiation strength. (bonus points if it has an optional event hook-in)

Reply:
What event would it raise?